muspinsim/input/keyword.py

```diff
@@ -51,6 +51,7 @@ class MuSpinKeyword(object):
     accept_range = True
     accept_as_x = False
     default = None
+    expr_size_bounds = (1, np.inf)
```

Comment:
Could you confirm for me how this is intended to behave? My understanding is this:
- that for simple text fields, it is a character limit
- for vector fields, it is a row-by-row limit on the number of elements present
- this does not limit the number of rows per keyword (that's validated elsewhere?)

Reply:
So the `MuSpinInput` class and the `KWClass` currently don't have any checks for the number parsed arguments for each row entered. 

I added `expr_size_bounds` which stores minimum and maximum number of parsed arguments (`LarkExpression` objects) allowed for that keyword

for keywords that only take one value - like `fitting_tolerance` has `expr_size_bounds=(1,1)` - which means it should only ever expect to receive 1 parsed argument. Same goes for `average_axes` - which can have many rows, but each row should only contain one argument

for fields like `orientation` we can get between 1-4 values so `expr_size_bounds=(1,4)`

It doesn't act as the character limit - I don't think there is a character limit for any keywords - so long as they get parsed into expected number of `LarkExpression` objects

Reply:
Incidentally 
From reading the documentation - all interaction keywords seem like they should only accept exactly 3 arguments for each row so `hyperfine` `zeeman`, `dipolar`, `quadrupolar` all should have `expr_size_bounds=(3,3)`. 

But for one of the tests in `test/test_config.py`, the `zeeman` keyword is only given 2 parameters
```
spins
    mu e
zeeman 1
    2 0
```
which is expected to pass - Is this a bug or should the program function when Zeeman  is given  2 arguments given instead of 3?

Reply:
Thanks, that makes a lot more sense to me now.

On the Zeeman thing, I think that has to be a bug. The docs say the zeeman coupling term specifies a local magnetic field, and I can't see how you would define that with only two elements - you need all three dimensions. (obviously for the `field` keyword we define a convention that allows you to get away with only one element, but that doesn't seem to be the case here)